process/rating/peerHonesty/peerScore_test.go

```diff
@@ -0,0 +1,418 @@
+		&mock.TimeCacheStub{},
+		nil,
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrNilCacher))
+}
+
+func TestNewP2pPeerHonesty_NilBlacklistedPkCacheShouldErr(t *testing.T) {
+	t.Parallel()
+
+	pph, err := NewP2pPeerHonesty(
+		createMockPeerHonestyConfig(),
+		nil,
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrNilBlackListedPkCache))
+}
+
+func TestNewP2pPeerHonesty_InvalidDecayCoefficientShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	cfg.DecayCoefficient = -0.1
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrInvalidDecayCoefficient))
+}
+
+func TestNewP2pPeerHonesty_InvalidDecayUpdateIntervalShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	cfg.DecayUpdateIntervalInSeconds = 0
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrInvalidDecayIntervalInSeconds))
+}
+
+func TestNewP2pPeerHonesty_InvalidMinScoreShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	cfg.MinScore = 1
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrInvalidMinScore))
+}
+
+func TestNewP2pPeerHonesty_InvalidMaxScoreShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	cfg.MaxScore = -1
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrInvalidMaxScore))
+}
+
+func TestNewP2pPeerHonesty_InvalidUnitValueShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	cfg.UnitValue = -1
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrInvalidUnitValue))
+}
+
+func TestNewP2pPeerHonesty_InvalidBadPeerThresholdShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	cfg.BadPeerThreshold = cfg.MinScore
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.True(t, check.IfNil(pph))
+	assert.True(t, errors.Is(err, process.ErrInvalidBadPeerThreshold))
+}
+
+func TestNewP2pPeerHonesty_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	cfg := createMockPeerHonestyConfig()
+	pph, err := NewP2pPeerHonesty(
+		cfg,
+		&mock.TimeCacheStub{},
+		&testscommon.CacherStub{},
+	)
+
+	assert.False(t, check.IfNil(pph))
+	assert.Nil(t, err)
+}
+
+func TestP2pPeerHonesty_Close(t *testing.T) {
```

Comment:
This test is not very short (in duration) - just a remark.

Reply:
I know, shortened